app/controllers/recipe_foods_controller.rb

```diff
@@ -0,0 +1,61 @@
+class RecipeFoodsController < ApplicationController
+  before_action :set_recipe_food, only: %i[show edit update destroy]
+
+  def index
+    @recipe_foods = RecipeFood.all
+  end
+
+  def new
+    @all_foods = Food.all
+    @recipe_food = RecipeFood.new
+  end
+
+  def edit
+    @all_foods = Food.all
+  end
+
+  def create
+    @recipe_food = RecipeFood.new(recipe_food_params)
+    recipe_id = params[:recipe_id]
+    @recipe_food.recipe_id = recipe_id
+
+    respond_to do |format|
+      if @recipe_food.save
+        format.html { redirect_to recipe_url(recipe_id), notice: 'Recipe food was successfully created.' }
+      else
+        format.html do
+          rendirect_to new_recipe_recipe_food_url(recipe_id), status: :unprocessable_entity,
+                                                              alert: 'Incomplete form submitted'
+        end
+      end
+    end
+  end
+
+  def update
+    respond_to do |format|
+      if @recipe_food.update(recipe_food_params)
+        format.html { redirect_to recipe_url(@recipe_food.recipe_id), notice: 'Recipe food was successfully updated.' }
+      else
+        format.html { render :edit, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  def destroy
```

Comment:
- [ ] Nicely done, kindly check when I click the delete button on the recipe page it throws the error in the screenshot below.
![error-receeipe](https://user-images.githubusercontent.com/48371156/229289842-1e6fd4ce-239f-4196-bf8b-2a1f80eace42.png)